kafka_schemas.go

```diff
@@ -0,0 +1,215 @@
+
+// GetConfig gets a Kafka Schema configuration
+func (h *KafkaSchemaHandler) GetConfig(project, service string) (*KafkaSchemaConfigResponse, error) {
+	path := buildPath("project", project, "service", service, "kafka", "schema", "config")
+	bts, err := h.client.doGetRequest(path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var r KafkaSchemaConfigResponse
+	errR := checkAPIResponse(bts, &r)
+
+	return &r, errR
+}
+
+// GetSubjects gets a list of Kafka Schema Subjects configuration
+func (h *KafkaSchemaHandler) GetSubjects(project, service string) (*KafkaSchemaSubjectsResponse, error) {
+	path := buildPath("project", project, "service", service, "kafka", "schema", "subjects")
+	bts, err := h.client.doGetRequest(path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var r KafkaSchemaSubjectsResponse
+	errR := checkAPIResponse(bts, &r)
+
+	return &r, errR
+}
+
+// GetSubjectVersions gets a Kafka Schema Subject versions
+func (h *KafkaSchemaHandler) GetSubjectVersions(project, service, name string) (*KafkaSchemaSubjectVersionResponse, error) {
+	path := buildPath("project", project, "service", service, "kafka", "schema", "subjects", name, "versions")
+	bts, err := h.client.doGetRequest(path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var r KafkaSchemaSubjectVersionResponse
+	errR := checkAPIResponse(bts, &r)
+
+	return &r, errR
+}
+
+// DeleteSubjectVersions deletes a Kafka Schema Subject versions
+func (h *KafkaSchemaHandler) DeleteSubjectVersions(project, service, name string, version int) error {
+	path := buildPath("project", project, "service", service, "kafka", "schema", "subjects", name, "versions", strconv.Itoa(version))
+	bts, err := h.client.doDeleteRequest(path, nil)
+	if err != nil {
+		return err
+	}
+
+	return checkAPIResponse(bts, nil)
+}
+
+// GetSubject gets a Kafka Schema Subject
+func (h *KafkaSchemaHandler) GetSubject(project, service, name string, version int) (*KafkaSchemaSubjectResponse, error) {
+	path := buildPath("project", project, "service", service, "kafka", "schema", "subjects", name, "versions", strconv.Itoa(version))
+	bts, err := h.client.doGetRequest(path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var r KafkaSchemaSubjectResponse
+	errR := checkAPIResponse(bts, &r)
+
+	return &r, errR
+}
+
+// ValidateSchema validates Kafka Schema
+func (h *KafkaSchemaHandler) ValidateSchema(
+	project, service, name string,
+	version int,
+	subject KafkaSchemaSubject) (bool, error) {
+	path := buildPath("project", project, "service", service, "kafka", "schema", "compatibility", "subjects", name, "versions", strconv.Itoa(version))
+
+	bts, err := h.client.doPostRequest(path, subject)
+	if err != nil {
+		return false, err
+	}
+
+	var r KafkaSchemaValidateResponse
+	errR := checkAPIResponse(bts, &r)
+
+	return r.IsCompatible, errR
+}
+
+// AddSubject adds a new kafka Schema
+func (h *KafkaSchemaHandler) AddSubject(project, service, name string, subject KafkaSchemaSubject) (*KafkaSchemaSubjectResponse, error) {
```

Comment:
need to think about these names, AddSubject is a bad name for something that actually Adds a schema

Reply:
sounds good